src/ring_theory/ideal/local_ring.lean

```diff
@@ -32,53 +32,43 @@ universes u v w
 
 /-- A commutative ring is local if it has a unique maximal ideal. Note that
   `local_ring` is a predicate. -/
-class local_ring (R : Type u) [comm_ring R] extends nontrivial R : Prop :=
-(is_local : ∀ (a : R), (is_unit a) ∨ (is_unit (1 - a)))
+class local_ring (R : Type u) [comm_semiring R] extends nontrivial R : Prop :=
+(nonunits_add : ∀ {a b : R}, a ∈ nonunits R → b ∈ nonunits R → a + b ∈ nonunits R)
 
 namespace local_ring
 
-variables {R : Type u} [comm_ring R] [local_ring R]
+variables {R : Type u}
```

Comment:
I think it is more readable if you put `[comm_semiring R] [local_ring R]` here., and use another variable `S` with `[comm_ring S] [local_ring S]`.